[#68] 검색결과에 따른 컬렉션뷰 구현 및 임시 배치

gustn3965 opened the pull request:
## 개요
- 검색결과에 따른 컬렉션뷰 구현 및 임시 배치 

## 시연 
![collec](https://user-images.githubusercontent.com/48749182/135959438-9d02207a-a3fe-4736-ade0-e1379f4ac978.gif)


## 작업 사항
- `CollectionView 섹션` 모델 생성
    - `CollectionView datasource` 및 `delegate`에서 쉽게 찾기 위함. 추후 변경에 있어서도 용이하다고 생각함. 
- `ContentsDetailColelcitonViewCell` 구현
    - 검색화면 결과에서 `글내용` 섹션에 해당하는 cell을 구현한다
- `LeftLabelRightButtonHeaderView` 구현
    - 좌측엔 타이틀, 서브타이틀, 우측엔 버튼을 가지는 뷰
- `SearchResultsViewController` 임시구현
    - 검색결과에 따른 결과를 갖는 `collectionView`를 배치
    - `SearchViewController`에서 `containerView` 를 이용하여 `SearchResultsViewController` 사용할 예정

## 테스트코드 
- `SceneDelegate` - `SearchResultsViewController` 로 변경하면 확인 가능합니다.

## 예외사항 
- 실제 데이터를 이용하지 않음.
- 더보기 버튼 구현하지 않음.

### Linked Issue

close #68

gustn3965 committed:
[#68] feat: CollectionView 섹션 모델 생성

gustn3965 committed:
[#68] view: ContentsDetailColelcitonViewCell 구현
- 검색화면 결과에서 - 글내용 섹션에 해당하는 cell을 구현한다

gustn3965 committed:
[#68] view: LeftLabelRightButtonHeaderView 구현
- 좌측엔 타이틀, 서브타이틀, 우측엔 버튼을 가지는 뷰

gustn3965 committed:
[#68] SearchResultsViewController 임시구현
- 검색결과에 따른 결과를 갖는 collectionView를 배치

jwonyLee commented on ThingLog/Model/ResultCollectionSection.swift:
```diff
@@ -0,0 +1,70 @@
+        "header"
+    }
+    
+    static func resultsCollectionViewLayout() -> UICollectionViewCompositionalLayout {
+        let layout: UICollectionViewCompositionalLayout = UICollectionViewCompositionalLayout { (sectionIndex: Int, _ : NSCollectionLayoutEnvironment) -> NSCollectionLayoutSection in
+            let groupHeight: NSCollectionLayoutDimension = sectionIndex == ResultCollectionSection.contents.section ? .absolute(110) :  .absolute(124)
```
🤔 모델에서 뷰의 레이아웃까지 만드는 건 조금... 고민해보면 좋을 거 같아욤

gustn3965 replied:
저도 고민하다가 여기에 넣었는데용! 그 이유는 `CollectionView`의 `section`을 이 모델에서 알 수 있고, 또 `section`별로 `layout `요소가 다를 수 있기 때문에 연관성이 있다고 판단하여 여기에 넣었습니당 🥲

jwonyLee commented on ThingLog/View/HeaderView/LeftLabelRightButtonHeaderView.swift:
```diff
@@ -0,0 +1,92 @@
+        ])
+    }
+    
+    func updateTitle(title: String?, subTitle: String? = nil ) {
+        titleLabel.text = title
+    }
```
🛠 `subTitle`을 매개변수로 받고서 `subTitleLabel.text`에 반영해주는 코드가 없어요!
```swift
subTitleLabel.text = "\(subTitle ?? "0") 건"
```

gustn3965 replied:
아아, 맞습니다. 반영하겠습니당!

jwonyLee reviewed (approved):
LGTM 👍 

:eyes: 화면 설계서랑 구현하신 부분이 쪼오금 다른데 알고 계신가욤?_?

| ContentsDetailCollectionViewCell | 화면 설계서 |
| -- | -- |
| <img width="393" alt="ContentsDetailCollectionViewCell" src="https://user-images.githubusercontent.com/15073405/135872485-7cce943a-84dd-43a9-824d-ad2553df5de5.png"> | <img width="460" alt="화면 설계서" src="https://user-images.githubusercontent.com/15073405/135872545-97141d29-5c4d-41b3-b4fe-5480f146e0be.png"> |

gustn3965 commented:
그리고 말씀해주신 화면설계서가 빠져있네요! 나름 계속 봤는데 놓쳐버렸네요 ㅠ  반영해서 올릴게요!

gustn3965 committed:
[#68] ref: 지원님 피드백 반영
- ContentsDetialCollecitonViewCell 수정( 화면설계서 디테일 놓친거 추가함 )
- LeftLabelRightButtonHeaderView 메서드 수정
